app/components/sidebars/main/main_sidebar.js

```diff
@@ -35,6 +37,7 @@ export default class ChannelSidebar extends Component {
             setChannelLoading: PropTypes.func.isRequired,
         }).isRequired,
         blurPostTextBox: PropTypes.func.isRequired,
+        channelIdsWithPosts: PropTypes.array,
```

Comment:
This is going to cause the sidebar to re-render a lot when it doesn't need to. Maybe we can make a view action to do this without the extra prop? If you make something like this, you can pass it into the MainSidebar's mapDispatchToProps, it won't re-render every time we load more posts.
```javascript
export function logChannelSwtich(channelId, currentChannelId) {
    return (dispatch, getState) => {
        if (channelId === currentChannelId) {
            return;
        }

        const metrics = [];
        if (getState().entities.channels.postsInChannel[channelId]) {
            metrics.push('channel:switch_loaded');
        } else {
            metrics.push('channel:switch_initial');
        }

        telemtry.reset();
        telemetry.start(metrics);
    };
}
````

Reply:
Got it, now updated using this.